src/server/server/frontendstructs.py

```diff
@@ -103,19 +103,21 @@ class UserComputedProfile(object):
     Represent the information about the user profile:
     """
     @staticmethod
-    def make_from_scratch(feature_vector, model_data):
-        return UserComputedProfile(feature_vector, model_data, datetime=None)
+    def make_from_scratch(initial_feature_vector, feature_vector, model_data):
+        return UserComputedProfile(initial_feature_vector, feature_vector, model_data, datetime=None)
 
     @staticmethod
-    def make_from_db(feature_vector, model_data, datetime):
-        return UserComputedProfile(feature_vector, model_data, datetime)
+    def make_from_db(initial_feature_vector, feature_vector, model_data, datetime):
+        return UserComputedProfile(initial_feature_vector, feature_vector, model_data, datetime)
 
```

Comment:
not fan of the naming "initial_feature_vector". It happens that we use this vector as an "initial" value, but we could use something else to compute this "initial" vector (and still need this vector), and we already use it after the initial phase. One last argument, it could be modified after "initial" phase.
For me it's more something like "direct_input_vector" or "explicit_feedback_vector" (besides positive and negative_feedback in modelData)